### DISCUSSION TITLE
How to create a user using CLI in built-in database

### DISCUSSION THREAD

**Discussion by alexe100** [emqx/emqx#13808] [General]
I want to store uses in built-in database and use ACLs to authorize them:

like this:

authentication = [
  {
   backend = "built_in_database"
   mechanism = "password_based"
   password_hash_algorithm {
      name = "sha256",
      salt_position = "suffix"
   }
   user_id_type = "username"
}
]

then..
add ACLs...

%% Allow MQTT client using username "dashboard"  to subscribe to "$SYS/#" topics
{allow, {user, "dashboard"}, subscribe, ["$SYS/#"]}.

%% Allow users with IP address "127.0.0.1" to publish/subscribe to topics "$SYS/#", "#"
{allow, {ipaddr, "127.0.0.1"}, all, ["$SYS/#", "#"]}.

%% Deny "All Users" subscribe to `$SYS/#`, `#` and `+/#`
{deny, all, subscribe, ["$SYS/#", {eq, "#"}, {eq, "+/#"}]}.

%% Allow any other publish/subscribe operation
%% NOTE: In production, change the last rule to `{deny, all}`, and set config: `authorization.no_match = deny`
{allow, all}.

But I am missing the step to create users into the built-in database. I am not using dashboard so I have to use CLI.
Thank you

**Comment by id** [comment]
Hi @alexe100 .

There is no CLI to create users, but you can use API:

```
curl -u api_key:api_secret \
  -X POST 'localhost:18083/api/v5/authentication/password_based:built_in_database/users' \
  -d '{"user_id": "foo", "password": "bar"}' \
  -H 'content-type: application/json'
```
